cmd/virt-launcher/virt-launcher.go

```diff
@@ -238,6 +240,13 @@ func main() {
 
 	log.InitializeLogging("virt-launcher")
 
+	// Block until all requested hookSidecars are ready
+	hookManager := hooks.GetManager()
+	err := hookManager.Collect(*requestedHooks)
```

Comment:
this needs the ability to timeout

Reply:
I'm making it so grpc Dials are 1s timeout, and the Collect function will have it configurable. Definitely can be improved with shared context timeout for all.